google-cloud-apis/java-apis/maven-apis/src/com/google/cloud/tools/intellij/cloudapis/maven/CloudLibraryDependencyWriter.java

```diff
@@ -101,44 +103,55 @@ private static void addLibrariesToMavenModule(
       @NotNull Set<CloudLibrary> libraries, @NotNull Module module, @Nullable String bomVersion) {
     Project project = module.getProject();
     MavenProject mavenProject = MavenProjectsManager.getInstance(project).findProject(module);
-    MavenDomProjectModel model =
-        MavenDomUtil.getMavenDomProjectModel(project, mavenProject.getFile());
+    MavenDomProjectModel model;
+    if (mavenProject != null) {
+      model = MavenDomUtil.getMavenDomProjectModel(project, mavenProject.getFile());
+    } else {
+      LOG.warn(
+          "Expected mavenized module "
+              + module
+              + " to have an associated MavenProject, but null was found. Possibly due to a missing pom.xml file.");
+      notifyFailedDependencies(project);
+      return;
+    }
 
-    new WriteCommandAction(project, DomUtil.getFile(model)) {
-      @Override
-      protected void run(@NotNull Result result) {
-        List<MavenDomDependency> dependencies = model.getDependencies().getDependencies();
-        Map<Boolean, List<MavenId>> mavenIdsMap =
-            libraries
-                .stream()
-                .map(CloudLibraryUtils::getFirstJavaClientMavenCoordinates)
-                .filter(Optional::isPresent)
-                .map(Optional::get)
-                .map(MavenUtils::toMavenId)
-                .collect(
-                    Collectors.partitioningBy(
-                        mavenId -> MavenUtils.isMavenIdInDependencyList(mavenId, dependencies)));
+    WriteCommandAction.writeCommandAction(project, DomUtil.getFile(model))
```

Comment:
this was a 1-to-1 refactoring to remove the call to deprecated API. the logic in the run lambda should be unchanged